external/kafka-0-10-sql/src/main/scala/org/apache/spark/sql/kafka010/KafkaSourceProvider.scala

```diff
@@ -353,13 +354,15 @@ private[kafka010] class KafkaSourceProvider extends DataSourceRegister
   }
 
   class KafkaTable(strategy: => ConsumerStrategy) extends Table
-    with SupportsMicroBatchRead with SupportsContinuousRead with SupportsStreamingWrite {
+    with SupportsRead with SupportsWrite {
```

Comment:
I don't necessarily think we should go back to the design, but it is a bit weird in context to see that SupportsRead and SupportsWrite end up being super-capabilities configured differently than normal capabilities.

Reply:
I raised the same concern before. @rdblue shall we reconsider it? I don't think we need the flexibility to change the read/write API(it will be a breaking change anyway). It's more important to make the API consistent with itself, and only has a single capability API.

Reply:
I don't think of these as super-capabilities, I think of them as a link between the v2 catalog API and the v2 source API. Using a trait like `SupportsRead` means we can later make changes to either one and continue to use the other. For example, if we added a `Stream` abstraction instead of using `Table` for both, then we would be able to have `Stream` implement `SupportsRead`. Does that make sense?

Reply:
My understanding is, the `Table` interface is the link between v2 source API and v2 catalog API. If we want to change the abstraction later, I think that will be very painful, moving out the read/write methods to separated interfaces doesn't help much.

Reply:
I agree with @jose-torres. The current mix is very confusing. Difficult to know what to use.